homeassistant/components/google_assistant/trait.py

```diff
@@ -824,16 +827,32 @@ class TemperatureControlTrait(_Trait):
 
     name = TRAIT_TEMPERATURE_CONTROL
 
+    commands = [
+        COMMAND_SET_TEMPERATURE,
+    ]
+
     @staticmethod
     def supported(domain, features, device_class, _):
         """Test if state is supported."""
         return (
             domain == sensor.DOMAIN
             and device_class == sensor.SensorDeviceClass.TEMPERATURE
-        )
+        ) or (domain == water_heater.DOMAIN)
 
     def sync_attributes(self):
         """Return temperature attributes for a sync request."""
+        domain = self.state.domain
+        if domain == water_heater.DOMAIN:
+            attrs = self.state.attributes
+            return {
+                "temperatureUnitForUX": _google_temp_unit(
+                    self.hass.config.units.temperature_unit
+                ),
+                "temperatureRange": {
+                    "minThresholdCelsius": attrs[water_heater.ATTR_MIN_TEMP],
```

Comment:
Climate has these attributes too, and them not being used is a bug (that can be fixed in another PR)

Reply:
Good point, I'd be happy to prepare/review that as well